File: app/src/main/java/org/scp/gymlog/ui/common/components/PerfectSquarePieChartView.java
package org.scp.gymlog.ui.common.components;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.mikephil.charting.charts.PieChart;

public class PerfectSquarePieChartView extends PieChart {

    public PerfectSquarePieChartView(@NonNull Context context) {
        super(context);
    }

    public PerfectSquarePieChartView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public PerfectSquarePieChartView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

        int width = getMeasuredWidth();
        setMeasuredDimension(width, width);
    }
}

File: app/src/main/java/org/scp/gymlog/ui/common/components/PerfectSquarePieChartView.kt
package org.scp.gymlog.ui.common.components

import android.content.Context
import android.util.AttributeSet
import com.github.mikephil.charting.charts.PieChart

class PerfectSquarePieChartView : PieChart {
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)

        val width = measuredWidth
        setMeasuredDimension(width, width)
    }
}